static/js/integration_bot_widget.js

```diff
@@ -0,0 +1,62 @@
+var integration_bot_widget = (function () {
+
+    var exports = {};
+
+    function create_bot(full_name, short_name, bot_avatar_file) {
+        var formData = new FormData();
+        var bot_api_key;
+        var bot_email;
+        formData.append('csrfmiddlewaretoken', csrf_token);
+        formData.append('full_name', full_name);
+        formData.append('short_name', short_name);
+        formData.append('file-'+0, bot_avatar_file);
+        channel.post({
+            url: '/json/bots',
+            data: formData,
+            cache: false,
+            processData: false,
+            contentType: false,
+            success: function (resp, statusText, xhr) {
+                var result = JSON.parse(xhr.responseText);
+                bot_api_key = result.api_key;
+                bot_email = result.email;
+            },
+            error: function (xhr) {
+                // UI yet to be creates with this div id
+                $('#bot_widget_error').text(JSON.parse(xhr.responseText).msg).show();
+            },
+        });
+        return { bot_api_key: bot_api_key,
+                 bot_email: bot_email,
+                };
```

Comment:
This `return` statement won't work, due to the asynchronous nature of javascript.  Instead, you should use a callback.  Someone on `#development help` can provide advice on restructuring this if needed.